### PR TITLE
FMWK-116 Support for query with filtering and response parsing

### PR DESCRIPTION
**Pull request by agrgr**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -113,6 +113,12 @@
             <version>${mockito.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>31.1-jre</version>
```
Comment (roimenashe) [inline pom.xml:119]: extract version to properties section.

File: src/main/java/com/aerospike/documentapi/data/DocumentFilterExp.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import com.aerospike.client.exp.Exp;
+import com.aerospike.documentapi.util.DocumentExp;
+
+public class DocumentFilterExp {
+
+    private Exp exp;
+
+    public DocumentFilterExp(String binName, String jsonPath, Operator.Simple operator, Object value) {
+        switch (operator) {
+
```
Comment (roimenashe) [inline src/main/java/com/aerospike/documentapi/data/DocumentFilterExp.java:12]: Unnecessary new line

File: src/main/java/com/aerospike/documentapi/data/DocumentFilterExp.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package com.aerospike.documentapi.data;
+
+import com.aerospike.client.exp.Exp;
+import com.aerospike.documentapi.util.DocumentExp;
+
+public class DocumentFilterExp {
+
+    private Exp exp;
```
Comment (roimenashe) [inline src/main/java/com/aerospike/documentapi/data/DocumentFilterExp.java:8]: Can be final? since we are setting it in the constructor.

Comment (agrgr) [reply src/main/java/com/aerospike/documentapi/data/DocumentFilterExp.java:8]: It is initialized in constructor. The constructor could have accepted just an Exp as a parameter which could be less controllable. Currently there is an enum parameter, and Exp can be built using each of enum values given.

**Review by reugn — APPROVED** [review]

**Pull request merged**